src/background/stores/global/user-configuration-store.ts

```diff
@@ -28,15 +28,21 @@ export class UserConfigurationStore extends BaseStoreImpl<UserConfigurationStore
     };
 
     constructor(
-        private readonly initialState: UserConfigurationStoreData,
+        private readonly persistedState: UserConfigurationStoreData,
         private readonly userConfigActions: UserConfigurationActions,
         private readonly indexDbApi: IndexedDBAPI,
     ) {
         super(StoreNames.UserConfigurationStore);
     }
 
+    private generateDefaultState(persisted: UserConfigurationStoreData): UserConfigurationStoreData {
+        const persistedState = cloneDeep(persisted);
+        const defaultState = cloneDeep(UserConfigurationStore.defaultState);
+        return Object.assign(defaultState, persistedState);
```

Comment:
We usually just use the spread operator; why not here?

Reply:
If there are common properties, for me it's more clear that assign() will take the value from the later one; Also assign has intellisense in the IDE that tell me which is source, will is target, while spread operator does not have that.